File: src/main/java/problems/FindDigits.java
package problems;

public class FindDigits {
    static int findDigits(int n) {
        int count = 0;
        StringBuilder stringBuilder = new StringBuilder(Integer.toString(n));
        for (int i = 0; i < stringBuilder.length(); i++) {
            char c = stringBuilder.charAt(i);
            int eachDigit = Integer.parseInt(String.valueOf(c));
            if(eachDigit !=0 && n % eachDigit == 0) {
             count = count + 1;   
            }
        }
        return count;
    }
}

File: js/find-digits.js
function findDigits(n) {
    let count = 0;
    const numToString = n.toString();
    for (let index = 0; index < numToString.length; index++) {
        const eachDigit = parseInt(numToString.charAt(index));

        if (eachDigit !== 0 && n % eachDigit === 0) {
            count = count + 1;
        }

    }
    return count;
}

// integer to string
const numToString = n.toString();

// String to integer
const num = parseInt(string);

// iterate over each character in string
for (let index = 0; index < numToString.length; index++) {
    const eachDigit = parseInt(numToString.charAt(index));
    // Do something
}
